test/common/ops/scoreTask.test.js

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable no-shadow */
+import scoreTask from '../../../common/script/ops/scoreTask';
+import {
+  generateUser,
+  generateDaily,
+  generateHabit,
+  generateTodo,
+  generateReward,
+} from '../../helpers/api-integration/v3';
+import common from '../../../common';
+
+let EPSILON = 0.0001; // negligible distance between datapoints
+
+/* Helper Functions */
+let rewrapUser = (user) => {
+  user._wrapped = false;
+  common.wrap(user);
+  return user;
+};
+
+let beforeAfter = async () => {
+  let before = await generateUser();
+  let after = _.cloneDeep(before);
+  rewrapUser(after);
+
+  return {
+    before,
+    after,
+  };
+};
+
+let expectGainedPoints = (before, after, beforeTask, afterTask) => {
+  expect(after.stats.hp).to.eql(50);
+  expect(after.stats.exp).to.be.greaterThan(before.stats.exp);
+  expect(after.stats.gp).to.be.greaterThan(before.stats.gp);
+  expect(afterTask.value).to.be.greaterThan(beforeTask.value);
+  if (afterTask.type === 'habit') {
+    expect(afterTask.history).to.have.length(1);
+  }
+};
+
+let expectClosePoints = (before, after, beforeTask, task) => {
+  expect(Math.abs(after.stats.exp - before.stats.exp)).to.be.lessThan(EPSILON);
+  expect(Math.abs(after.stats.gp - before.stats.gp)).to.be.lessThan(EPSILON);
+  expect(Math.abs(task.value - beforeTask.value)).to.be.lessThan(EPSILON);
+};
+
+let _expectRoughlyEqualDates = (date1, date2) => {
+  expect(date1.toString()).to.eql(date2.toString());
+};
+
+describe('shared.ops.scoreTask', () => {
+  let ref;
+
+  beforeEach(async () => {
+    ref = await beforeAfter();
+  });
+
+  it('throws error not enough gold', async () => {
```

Comment:
This should be a sentence and should fully explain the scenario it is testing.

`it throws an error when scoring a reward if user does not have enough gold`

If there are a couple of reward related assertions, you can wrap it all in a `context('Rewards')` and then the it would be `it('thows error if the user does not have enough gold')`